scipy/stats/_multicomp.py

```diff
@@ -0,0 +1,311 @@
+        +- the allowance.
+
+        Parameters
+        ----------
+        confidence_level : float, optional
+            Confidence level for the computed confidence interval
+            of the estimated proportion. Default is .95.
+
+        Returns
+        -------
+        ci : ``ConfidenceInterval`` object
+            The object has attributes ``low`` and ``high`` that hold the
+            lower and upper bounds of the confidence intervals for each
+            comparison. The high and low values are accessible for each
+            comparison at index ``(i, j)`` between groups ``i`` and ``j``.
+
+        """
+        allowance = self._allowance(confidence_level=confidence_level)
+        diff_means = self._observations_mean - self._control_mean
+        return ConfidenceInterval(
+            low=diff_means-allowance,
+            high=diff_means+allowance
+        )
+
+
+def dunnett(
+    *observations: npt.ArrayLike, control: npt.ArrayLike,
+    alternative: Literal['two-sided', 'less', 'greater'] = "two-sided",
+    random_state: SeedType = None
+) -> DunnettResult:
+    """Dunnett's test.
+
+    Parameters
+    ----------
+    observations1, observations2, ... : 1D array_like
+        The sample measurements for each experiment group.
+    control : 1D array_like
+        The sample measurements for the control group.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available (default is 'two-sided'):
+
+        * 'two-sided': the means of the distributions underlying the samples
+          are unequal.
+        * 'less': the mean of the distribution underlying the first sample
+          is less than the mean of the distribution underlying the second
+          sample.
+        * 'greater': the mean of the distribution underlying the first
+          sample is greater than the mean of the distribution underlying
+          the second sample.
+    random_state : {None, int, `numpy.random.Generator`}, optional
+        If `random_state` is an int or None, a new `numpy.random.Generator` is
+        created using ``np.random.default_rng(random_state)``.
+        If `random_state` is already a ``Generator`` instance, then the
+        provided instance is used.
+
+    Returns
+    -------
+    res : DunnettResult
+        An object containing attributes:
+
+        statistic : scalar or ndarray
+            The z-score for the test.  For 1-D inputs a scalar is
+            returned.
+        pvalue : scalar ndarray
+            The p-value for the test.
+
+    See Also
+    --------
+    tukey_hsd : performs pairwise comparison of means.
+
+    Notes
+    -----
+    Dunnett's test [1]_ compares the mean of multiple experiment groups against
+    a control group. `tukey_hsd` instead, performs pairwise comparison of
+    means. It means Dunnett's test performs less tests making it more powerful.
```

Comment:
Performing fewer tests does not necessarily make it more powerful. There is a missing link here - because it performs fewer tests, **there is less p-value adjustment**, and this is what results in higher power.

Reply:
I had Benjamini-Hochberg on the mind when I wrote this. I'll need to reconsider. Sorry about that.